homeassistant/components/esphome/config_flow.py

```diff
@@ -250,6 +252,41 @@ async def async_step_zeroconf(
 
         return await self.async_step_discovery_confirm()
 
+    async def async_step_mqtt(
+        self, discovery_info: MqttServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle MQTT discovery."""
+        device_info: dict[str, Any] = json_loads(discovery_info.payload)
```

Comment:
You could narrow this with `json_loads_object` instead and drop the type

Reply:
sounds good, done